samples/OpenIdConnectSample/Startup.cs

```diff
@@ -39,6 +40,23 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
 
             app.Run(async context =>
             {
+                var authContext = new AuthenticateContext(CookieAuthenticationDefaults.AuthenticationScheme);
+                await context.Authentication.AuthenticateAsync(authContext);
+                if (context.Request.Path == new PathString("/remote-sign-out"))
+                {
+                    var session = authContext.Properties[OpenIdConnectSessionProperties.SessionState];
+                    var sid = context.Request.Query["sid"];
+                    if (string.Equals(sid, session, System.StringComparison.Ordinal))
```

Comment:
the contract is: if discover.logout_session_supported==true, then a 'sid' claim should show up in the id_token. I agree that session_state is a different thing, however I suspect the that it may be the same. It's confusing that the two 'logouts' have a different model for the session identifier, found in two different places.
My take is that AAD is not supporting the 'sid' yet. We added a property to support this in OIDCConfiguration - HttpLogoutSupported, I think RC1. Not sure, but it is there now. You can just check the metadata.